src/view/tst/ViewApiFunctionalityTest.cpp

```diff
@@ -295,6 +295,88 @@ TEST_F(ViewApiFunctionalityTest, AddGetSetCurrentRemoveAll)
+
+        timestamp += TEST_VIEW_ITEM_DURATION;
+    }
+
+    // content view should be full now, can no frame should be dropped yet.
+    EXPECT_EQ(0, gCallCount);
+
+    // should trigger a fragment getting dropped
+    EXPECT_EQ(STATUS_SUCCESS, contentViewAddItem(mContentView,
+                                                 timestamp,
+                                                 timestamp,
+                                                 TEST_VIEW_ITEM_DURATION,
+                                                 INVALID_ALLOCATION_HANDLE_VALUE,
+                                                 0,
+                                                 VIEW_ITEM_ALLOCAITON_SIZE,
+                                                 i % KEY_FRAME_INTERVAL == 0 ? ITEM_FLAG_FRAGMENT_START : ITEM_FLAG_NONE));
+    // entire fragment should get dropped
+    EXPECT_EQ(gCallCount, KEY_FRAME_INTERVAL);
+}
+
+TEST_F(ViewApiFunctionalityTest, overflowItemCountWithDropUntilFragmentStart)
+{
+    const UINT64 KEY_FRAME_INTERVAL = 25;
+    const UINT64 FPS = 25;
+    const UINT64 TEST_VIEW_ITEM_DURATION = HUNDREDS_OF_NANOS_IN_A_SECOND / FPS;
+    const UINT64 TEST_MAX_BUFFER_DURATION = 2 * HUNDREDS_OF_NANOS_IN_A_SECOND;
+    UINT64 timestamp = 0;
+    UINT32 i;
+
+    // max view item count will run out first
+    CreateContentView(DROP_UNTIL_FRAGMENT_START, 2 * FPS, 100 * HUNDREDS_OF_NANOS_IN_A_SECOND);
+
+    for(i = 0; i < 2 * FPS; ++i) {
+        EXPECT_EQ(STATUS_SUCCESS, contentViewAddItem(mContentView,
+                                                     timestamp,
+                                                     timestamp,
+                                                     TEST_VIEW_ITEM_DURATION,
+                                                     INVALID_ALLOCATION_HANDLE_VALUE,
+                                                     0,
+                                                     VIEW_ITEM_ALLOCAITON_SIZE,
+                                                     i % KEY_FRAME_INTERVAL == 0 ? ITEM_FLAG_FRAGMENT_START : ITEM_FLAG_NONE));
+
+        timestamp += TEST_VIEW_ITEM_DURATION;
+    }
+
+    // content view should be full now, can no frame should be dropped yet.
+    EXPECT_EQ(0, gCallCount);
+
+    // should trigger a fragment getting dropped
+    EXPECT_EQ(STATUS_SUCCESS, contentViewAddItem(mContentView,
+                                                 timestamp,
+                                                 timestamp,
+                                                 TEST_VIEW_ITEM_DURATION,
+                                                 INVALID_ALLOCATION_HANDLE_VALUE,
+                                                 0,
+                                                 VIEW_ITEM_ALLOCAITON_SIZE,
+                                                 i % KEY_FRAME_INTERVAL == 0 ? ITEM_FLAG_FRAGMENT_START : ITEM_FLAG_NONE));
+    // entire fragment should get dropped
+    EXPECT_EQ(gCallCount, KEY_FRAME_INTERVAL);
+}
```

Comment:
Could you add the following test cases 1) to drop entire content view on overflow with the new strategy and 2) drop until new session 3) new strategy with EoFR.

Reply:
test added.